lib/vicadmin/validate.go

```diff
@@ -62,6 +63,47 @@ const (
 	BadStatus  = template.HTML(`<i class="icon-attention"></i>`)
 )
 
+func GetMgmtIP() net.IPNet {
+	var mgmtIP net.IPNet
+	// management alias may not be present, try others if not found
+	link := LinkByNameOrAlias("management")
+	if link == nil {
+		link = LinkByNameOrAlias("external")
+	}
+	if link == nil {
+		link = LinkByNameOrAlias("client")
+	}
+	if link == nil {
+		log.Error("unable to find any interfaces when searching for mgmt IP")
+		return mgmtIP
+	}
+
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		log.Errorf("error getting address list: %s", err)
+		return mgmtIP
+	}
+	if len(addrs) == 0 {
+		log.Warnf("no addresses on interface when searching for mgmt IP")
+		return mgmtIP
+	}
+	if len(addrs) > 1 {
+		log.Warnf("multiple addresses on interface when searching for mgmt IP, using first")
+	}
+	return *addrs[0].IPNet
+}
+
+func LinkByNameOrAlias(name string) netlink.Link {
+	l, err := netlink.LinkByName(name)
+	if l == nil {
+		l, err = netlink.LinkByAlias(name)
+		if err != nil {
+			log.Errorf("interface %s not found", name)
```

Comment:
are you sure this is an error? it seems like the caller here really tries to make several calls.
I also would pass just bunch of possible names at once, so the logic of the caller would be simplified.

```
 func LinkByOneOfNameOrAlias(name ...string) netlink.Link {
 	for _, n := range name {
 	    if l, _ := netlink.LinkByName(n); l != nil {
 	    	return l
 	    }
		if if l, _ := netlink.LinkByAlias(n); l != nil {
			return l
		}
	}
	return nil
}
```